dali/server/daserver.cpp

```diff
@@ -392,6 +392,42 @@ static StringBuffer &formatDaliRole(StringBuffer &out, unsigned __int64 role)
     return out.append(queryRoleName((DaliClientRole)role));
 }
 
+#ifdef _CONTAINERIZED
+static IPropertyTree * getContainerLDAPConfiguration(const IPropertyTree *appConfig)
+{
+    const char * authMethod = appConfig->queryProp("@auth");
+    if (streq(authMethod, "none"))
+    {
+        WARNLOG("ECLWatch is unsafe, no security manager specified in configuration (auth: none)");
+        return nullptr; //no security manager
+    }
+
+    if (!streq(authMethod, "ldap"))
+    {
+        throw makeStringExceptionV(-1, "Unrecognized auth method specified, (auth: %s)", authMethod);
+    }
+
+    const char *ldapAddress = appConfig->queryProp("@ldapAddress");
+    if (isEmptyString(ldapAddress))
+        throw makeStringException(-1, "LDAP not configured (missing 'ldapAddress').  To run without security set 'auth: none'");
+
+    //Get default LDAP attributes from ldap.yaml
+    StringBuffer ldapDefaultsFile(hpccBuildInfo.componentDir);
+    char sepchar = getPathSepChar(ldapDefaultsFile.str());
+    addPathSepChar(ldapDefaultsFile, sepchar).append("applications").append(sepchar).append("common").append(sepchar).append("ldap").append(sepchar).append("ldap.yaml");
+    Owned<IPropertyTree> defaults;
+    if (checkFileExists(ldapDefaultsFile))
+        defaults.setown(createPTreeFromYAMLFile(ldapDefaultsFile.str()));
+
+    //Build merged configuration
+    Owned<IPropertyTree> mergedConfig = defaults->getPropTree("ldap");
```

Comment:
if line 419 returned false, defaults could be null here.
I think line 419 should assert (throw an error) if the file is missing though?
createPTreeFromYAMLFile returns null if the file doesn't exist, so you can check that to avoid the checkFileExists, to avoid opening the file twice to check.